### PR TITLE
Functional tests - Add new test 'BO > Catalog > Filter products table v2'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add new test 'BO > Catalog > Filter products v2'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| Related PRs       | no
| How to test?      | no need
| Possible impacts? | no

### CODE REVIEW

File: tests/UI/campaigns/productV2/functional/01_filterProducts.ts [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+          const textColumn = await productsPage.getTextColumn(page, test.args.filterBy, i);
+
+          if (test.args.filterBy === 'id_product' || test.args.filterBy === 'price'
+            || test.args.filterBy === 'quantity') {
+            await expect(textColumn).to.within(test.args.filterValue.min, test.args.filterValue.max);
+          } else if (test.args.filterBy === 'active' && test.args.filterValue === 'Yes') {
+            await expect(textColumn).to.be.true;
+          } else {
+            await expect(textColumn).to.be.contain(test.args.filterValue);
+          }
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/01_filterProducts.ts]: Why not create a `test.args.filterResult` => 'boolean', 'text' or 'number' which use each check rather using the the `filterBy` key ?

**Review by Progi1984 — COMMENTED** [review]
A feedback :)

**Review by Progi1984 — CHANGES_REQUESTED** [review]
Need a rebase :)

File: tests/UI/campaigns/productV2/functional/01_filterProducts.ts (hunk 1 above)
Comment (nesrineabdmouleh) [reply tests/UI/campaigns/productV2/functional/01_filterProducts.ts]: Thank you @Progi1984 :)
Yes I thought about the new arg but the problem  in this case is  : we have different types of comparison (`.to.within`, `to.be.true`,` .to.be.contain` )
Do you have a solution for that ?

Comment (Progi1984) [reply tests/UI/campaigns/productV2/functional/01_filterProducts.ts]: Use  a switch case for each type and a default triggers an error.

File: tests/UI/campaigns/productV2/functional/01_filterProducts.ts [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+
+// Import pages
+import dashboardPage from '@pages/BO/dashboard';
+import productsPage from '@pages/BO/catalog/productsV2';
+
+// Import data
+import {Products} from '@data/demo/products';
+import {Categories} from '@data/demo/categories';
+
+const baseContext = 'productV2_functional_filterProducts';
```
Comment (Progi1984) [inline tests/UI/campaigns/productV2/functional/01_filterProducts.ts]:
```suggestion
const baseContext: string = 'productV2_functional_filterProducts';
```

**Review by Progi1984 — DISMISSED** [review]

**Review by SD1982 — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by boubkerbribri** [comment]
Thanks @nesrineabdmouleh

**Pull request merged**